diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
--- a/qtbase/src/corelib/global/qlibraryinfo.cpp
+++ b/qtbase/src/corelib/global/qlibraryinfo.cpp
@@ -106,21 +106,6 @@ static std::unique_ptr<QSettings> findConfiguration()
     QString qtconfig = QStringLiteral(":/qt/etc/qt.conf");
     if (QResource(qtconfig, QLocale::c()).isValid())
         return std::make_unique<QSettings>(qtconfig, QSettings::IniFormat);
-#ifdef Q_OS_DARWIN
-    CFBundleRef bundleRef = CFBundleGetMainBundle();
-    if (bundleRef) {
-        QCFType<CFURLRef> urlRef = CFBundleCopyResourceURL(bundleRef,
-                                                           QCFString("qt.conf"_L1),
-                                                           0,
-                                                           0);
-        if (urlRef) {
-            QCFString path = CFURLCopyFileSystemPath(urlRef, kCFURLPOSIXPathStyle);
-            qtconfig = QDir::cleanPath(path);
-            if (QFile::exists(qtconfig))
-                return std::make_unique<QSettings>(qtconfig, QSettings::IniFormat);
-        }
-    }
-#endif
     if (QCoreApplication::instance()) {
         QString pwd = QCoreApplication::applicationDirPath();
         qtconfig = pwd + u"/qt" QT_STRINGIFY(QT_VERSION_MAJOR) ".conf"_s;
