From 16167eb8f28f49b6fede5dbdb7d4e41fcac2fa07 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 14 May 2025 15:10:58 +0200
Subject: [PATCH] Remove need to set QT_HOST_PATH for cross-compiled builds

---
 cmake/QtConfigDependencies.cmake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/QtConfigDependencies.cmake.in b/cmake/QtConfigDependencies.cmake.in
index b006f052f7f..1d05bd9f47a 100644
--- a/qtbase/cmake/QtConfigDependencies.cmake.in
+++ b/qtbase/cmake/QtConfigDependencies.cmake.in
@@ -8,7 +8,7 @@ if(DEFINED QT_REQUIRE_HOST_PATH_CHECK)
 elseif(DEFINED ENV{QT_REQUIRE_HOST_PATH_CHECK})
     set(__qt_platform_requires_host_info_package "$ENV{QT_REQUIRE_HOST_PATH_CHECK}")
 elseif(CMAKE_CROSSCOMPILING OR DEFINED QT_HOST_PATH)
-    set(__qt_platform_requires_host_info_package "@platform_requires_host_info_package@")
+    set(__qt_platform_requires_host_info_package "FALSE")
 else()
     set(__qt_platform_requires_host_info_package FALSE)
 endif()
