diff --git a/src/quicktestutils/CMakeLists.txt b/src/quicktestutils/CMakeLists.txt
index 06f0ad25..25f8f382 100644
--- a/qtdeclarative/src/quicktestutils/CMakeLists.txt
+++ b/qtdeclarative/src/quicktestutils/CMakeLists.txt
@@ -31,6 +31,7 @@ qt_internal_add_module(QuickTestUtilsPrivate
     PUBLIC_LIBRARIES
         Qt::Core
         Qt::Test
+        Qt::Network
         Qt::Qml
         Qt::QmlPrivate
 )
