diff --git a/src/designer/src/components/lib/CMakeLists.txt b/src/designer/src/components/lib/CMakeLists.txt
index c4ebae39c..a6e3792da 100644
--- a/qttools/src/designer/src/components/lib/CMakeLists.txt
+++ b/qttools/src/designer/src/components/lib/CMakeLists.txt
@@ -98,6 +98,7 @@ qt_internal_add_module(DesignerComponentsPrivate
     NO_UNITY_BUILD_SOURCES
         ../tabordereditor/tabordereditor.cpp # redefinition of 'QMetaTypeId<QList<QWidget *>>' (from qdesigner_resource.cpp)
         ../formeditor/formwindow.cpp         # explicit specialization of 'QMetaTypeId<QWidget *>' after instantiation
+        ../formeditor/deviceprofiledialog.cpp
     DEFINES
         QDESIGNER_COMPONENTS_LIBRARY
         QT_STATICPLUGIN
