diff --git a/src/quick3d/CMakeLists.txt b/src/quick3d/CMakeLists.txt
index d768c732..7b0993aa 100644
--- a/qtquick3d/src/quick3d/CMakeLists.txt
+++ b/qtquick3d/src/quick3d/CMakeLists.txt
@@ -73,6 +73,7 @@ qt_internal_add_qml_module(Quick3D
         extensions/qquick3dextensionhelpers.cpp extensions/qquick3dextensionhelpers_p.h
     NO_UNITY_BUILD_SOURCES
         qquick3dspecularglossymaterial.cpp # redefinition of 'ensureNormalized' (from qquick3dprincipledmaterial.cpp)
+        qquick3dcustommaterial.cpp
     DEFINES
         QT_BUILD_QUICK3D_LIB
     LIBRARIES
