diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
index 1a8cc08de5..7c67bdd517 100644
--- a/qtbase/src/corelib/CMakeLists.txt
+++ b/qtbase/src/corelib/CMakeLists.txt
@@ -628,6 +628,7 @@ qt_internal_extend_target(Core CONDITION APPLE
     NO_UNITY_BUILD_SOURCES
         io/qfilesystemengine_unix.cpp # avoid undefined DT_* macros
         io/qlockfile_unix.cpp         # avoid undefined SOCK_MAXADDRLEN (in proc_info.h)
+        kernel/qelapsedtimer_mac.cpp  # avoid qt_nanosleep clash
 )
 
 qt_internal_extend_target(Core CONDITION QT_FEATURE_thread
