diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 0e04b722b4..f3cf558dd8 100644
--- a/qtbase/src/gui/CMakeLists.txt
+++ b/qtbase/src/gui/CMakeLists.txt
@@ -360,7 +360,7 @@ qt_internal_extend_target(Gui CONDITION QT_FEATURE_opengl
         rhi/qrhigles2_p_p.h
 )
 
-qt_internal_extend_target(Gui CONDITION MACOS
+qt_internal_extend_target(Gui CONDITION MACOS AND QT_FEATURE_opengl
     SOURCES
         platform/macos/qcocoanativeinterface.mm
     LIBRARIES
@@ -370,6 +370,18 @@ qt_internal_extend_target(Gui CONDITION MACOS
         ${FWAppKit}
 )
 
+qt_internal_extend_target(Gui CONDITION MACOS AND NOT QT_FEATURE_opengl
+    SOURCES
+        platform/macos/qcocoanativeinterface.mm
+    LIBRARIES
+        ${FWAppKit}
+        ${FWCarbon}
+    PUBLIC_LIBRARIES
+        ${FWAppKit}
+    DEFINES
+        QT_NO_OPENGL
+)
+
 qt_internal_extend_target(Gui CONDITION APPLE
     SOURCES
         image/qimage_darwin.mm
