diff --git a/qmake/CMakeLists.txt b/qmake/CMakeLists.txt
index 41bf2c77d9..8c09c22f2f 100644
--- a/qtbase/qmake/CMakeLists.txt
+++ b/qtbase/qmake/CMakeLists.txt
@@ -31,10 +31,14 @@ target_include_directories(QtLibraryInfo PUBLIC
 
 # Chop off the "/mkspecs" part of INSTALL_MKSPECSDIR
 get_filename_component(hostdatadir "${INSTALL_MKSPECSDIR}" DIRECTORY)
+message(STATUS "zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz1 QT_HOST_DATADIR=${hostdatadir}")
+
 if("${hostdatadir}" STREQUAL "")
     set(hostdatadir ".")
 endif()
 
+message(STATUS "zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz2 QT_HOST_DATADIR=${hostdatadir}")
+
 qt_internal_library_deprecation_level(deprecation_define)
 
 target_compile_definitions(QtLibraryInfo PUBLIC
diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index a8934fa38c..fb06e83456 100644
--- a/qtbase/src/corelib/global/qlibraryinfo.cpp
+++ b/qtbase/src/corelib/global/qlibraryinfo.cpp
@@ -457,20 +457,20 @@ QLibraryInfoPrivate::LocationInfo QLibraryInfoPrivate::locationInfo(QLibraryInfo
     static constexpr auto qtConfEntries = qOffsetStringArray(
         "Prefix", ".",
         "Documentation", "doc", // should be ${Data}/doc
-        "Headers", "include",
-        "Libraries", "lib",
+        "Headers", "Library/include/qt6",
+        "Libraries", "Library/lib",
 #ifdef Q_OS_WIN
-        "LibraryExecutables", "bin",
+        "LibraryExecutables", "Library/lib/qt6",
 #else
         "LibraryExecutables", "libexec", // should be ${ArchData}/libexec
 #endif
-        "Binaries", "bin",
+        "Binaries", "Library/lib/qt6",
         "Plugins", "plugins", // should be ${ArchData}/plugins
 
         "QmlImports", "qml", // should be ${ArchData}/qml
 
         "ArchData", ".",
-        "Data", ".",
+        "Data", "Library/lib/qt6",
         "Translations", "translations", // should be ${Data}/translations
         "Examples", "examples",
         "Tests", "tests"
