diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index a8934fa38c..523ac4f808 100644
--- a/qtbase/src/corelib/global/qlibraryinfo.cpp
+++ b/qtbase/src/corelib/global/qlibraryinfo.cpp
@@ -458,19 +458,19 @@ QLibraryInfoPrivate::LocationInfo QLibraryInfoPrivate::locationInfo(QLibraryInfo
         "Prefix", ".",
         "Documentation", "doc", // should be ${Data}/doc
         "Headers", "include",
-        "Libraries", "lib",
+        "Libraries", "Library/lib",
 #ifdef Q_OS_WIN
-        "LibraryExecutables", "bin",
+        "LibraryExecutables", "Library/lib/qt6",
 #else
         "LibraryExecutables", "libexec", // should be ${ArchData}/libexec
 #endif
-        "Binaries", "bin",
+        "Binaries", "Library/lib/qt6",
         "Plugins", "plugins", // should be ${ArchData}/plugins
 
         "QmlImports", "qml", // should be ${ArchData}/qml
 
         "ArchData", ".",
-        "Data", ".",
+        "Data", "Library/share/qt6",
         "Translations", "translations", // should be ${Data}/translations
         "Examples", "examples",
         "Tests", "tests"
