diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index a8934fa38c..3ff8be9290 100644
--- a/qtbase/src/corelib/global/qlibraryinfo.cpp
+++ b/qtbase/src/corelib/global/qlibraryinfo.cpp
@@ -401,6 +401,7 @@ static QString getRelocatablePrefix(QLibraryInfoPrivate::UsageMode usageMode)
                 + slash + QT_CONFIGURE_LIBLOCATION_TO_PREFIX_PATH
                 + slash + libdir
                 + slash + qtCoreImpLibFileName;
+	std::cout << "zzzzzzzzzzzzzzzzzz qtCoreImpLibPath="<<qtCoreImpLibPath.toStdString()<<std::endl;
         if (!QFileInfo::exists(qtCoreImpLibPath)) {
             // We did not find a corresponding import library and conclude that this is a
             // windeployqt'ed executable.
