diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
index 076eabcbe7..23e8adffb0 100644
--- a/qtbase/src/corelib/CMakeLists.txt
+++ b/qtbase/src/corelib/CMakeLists.txt
@@ -329,6 +329,7 @@ qt_internal_add_module(Core
     NO_PCH_SOURCES
         compat/removed_api.cpp
         global/qsimd.cpp
+	global/qassert.cpp
     DEFINES
         QT_NO_CONTEXTLESS_CONNECT
         QT_NO_FOREACH
