diff --git a/src/assistant/qhelpgenerator/CMakeLists.txt b/src/assistant/qhelpgenerator/CMakeLists.txt
index 7230d248e..ce2944f17 100644
--- a/qttools/src/assistant/qhelpgenerator/CMakeLists.txt
+++ b/qttools/src/assistant/qhelpgenerator/CMakeLists.txt
@@ -62,7 +62,7 @@ ${import_plugin_code}
         )
     endif()
 else() # QT_BUILD_SHARED_LIBS
-    if(QT_SUPERBUILD)
+	if(TARGET qhelpgenerator AND QT_SUPERBUILD)
         add_dependencies(qhelpgenerator ${needed_plugins})
     endif()
 endif()
