diff --git a/src/network/CMakeLists.txt b/src/network/CMakeLists.txt
index ec6e5f1426..c5636689b0 100644
--- a/qtbase/src/network/CMakeLists.txt
+++ b/qtbase/src/network/CMakeLists.txt
@@ -381,6 +381,8 @@ if (WIN32)
                                 PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON)
 endif()
 
+set_source_files_properties(socket/qnativesocketengine_unix.cpp PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON)
+
 # include the snippet projects for developer-builds
 if(QT_FEATURE_private_tests)
     add_subdirectory(doc/snippets/network)
